### DISCUSSION TITLE
Granular major and minor updates

### DISCUSSION THREAD

**Discussion by lsoica** [renovatebot/renovate#17360] [Request Help] [answered]
Is it possible to enable major updates for some managers/package patterns and disable for the others ?

E.g. I want to disable major upgrades for centos base image in a docker file (should stay on 7 instead of 8):
```
FROM centos:7.9.2009
```

but enable major upgrades for alpine packages inside a dockerfile (managed by a regex manager). I want this updates to 11
```
# renovate: datasource=repology depName=alpine_3_16/gcc versioning=loose
ENV GCC_VERSION=10.x.y
```

**Comment by viceice** [comment, marked as answer]
yes, use package rules

**Reply by secustor** [reply]
https://docs.renovatebot.com/configuration-options/#packagerules

**Comment by lsoica** [comment]
Thanks both. It worked with something like this as you suggested:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "major": {
    "enabled": false
  },
  "packageRules": [
    {
      "matchManagers": ["regex"],
      "major": {
        "enabled": true
      }    
    }
  ]
}
```
